packages/arb-token-bridge-ui/src/hooks/useRedeemRetryable.ts

```diff
@@ -24,10 +28,19 @@ export function useRedeemRetryable(): UseRedeemRetryableResult {
   } = useAppState()
   const {
     l1: { provider: l1Provider },
-    l2: { network: l2Network }
+    l2: { network: l2Network },
+    isConnectedToArbitrum
   } = useNetworksAndSigners()
-  const { data: signer } = useSigner()
+  const { data: l2Signer } = useSigner({
+    chainId: l2Network.chainID
+  })
   const l2NetworkName = getNetworkName(l2Network.chainID)
+  const { switchNetworkAsync } = useSwitchNetwork({
```

Comment:
Nice! Didn't realize we also had `switchNetworkAsync`. Looking through the codebase, should we also be using it in `TransferPanel.tsx` and `TransferPanelMain.tsx`? We are currently using `switchNetwork` which can't be awaited.